app/src/main/kotlin/xyz/block/augurref/service/MempoolCollector.kt

```diff
@@ -70,6 +71,49 @@ class MempoolCollector(
     return latestFeeEstimate.get()
   }
 
+  /**
+   * Get the latest fee estimate for estimated blocks
+   */
+  fun getLatestFeeEstimateForEstimatedBlocks(numOfBlocks: Double): FeeEstimate? {
+    val startTime = System.currentTimeMillis()
+    logger.debug("Collecting mempool data")
+
+    // Get height and mempool data directly in Augur format
+    val (blockHeight, transactions) = bitcoinClient.getHeightAndMempoolTransactions()
+    logger.debug("Got mempool data: ${transactions.size} transactions at height $blockHeight")
+
+    // Create snapshot
+    val mempoolSnapshot = MempoolSnapshot.fromMempoolTransactions(
+      transactions = transactions,
+      blockHeight = blockHeight,
+    )
+
+    // Save snapshot
+    // persistence.saveSnapshot(mempoolSnapshot)
+    // logger.info("Mempool snapshot saved: ${transactions.size} transactions at height $blockHeight")
+
+    // Fetch the last day's snapshots
+    logger.debug("Fetching snapshots from the last day")
+    val lastDaySnapshots = persistence.getSnapshots(
+      LocalDateTime.now().minusDays(1),
+      LocalDateTime.now(),
+    )
+    logger.debug("Retrieved ${lastDaySnapshots.size} snapshots from the last day")
+
+    if (lastDaySnapshots.isNotEmpty()) {
+      // Calculate fee estimate for x blocks
+      logger.debug("Calculating fee estimates")
+      val newFeeEstimate = feeEstimator.calculateEstimatesForNumberOfBlocks(lastDaySnapshots, numOfBlocks)
+      return newFeeEstimate
+    } else {
+      logger.warn("No snapshots available for fee estimation")
+    }
+    return FeeEstimate(
+      estimates = emptyMap(),
+      timestamp = Instant.EPOCH,
+    )
```

Comment:
We might want to return an error message or "No snapshots available for fee estimation"